infra.bs

```diff
@@ -258,11 +262,31 @@ inclusive.
 
 <h3 id=strings>Strings</h3>
 
-<p>A <dfn export>string</dfn> is a sequence of <a>code points</a>. Strings are denoted by double
-quotes and monospace font.
+<p>A <dfn export>JavaScript string</dfn> is a sequence of unsigned 16-bit integers, also known as
+<dfn export lt="code unit">code units</dfn>. A <a>JavaScript string</a> can also be interpreted as
+containing <a>code points</a>, per the conversion defined in <a>The String Type</a> section of the
+JavaScript specification. [[!ECMA-262]]
+
+<p>A <dfn export>scalar value string</dfn> is a sequence of <a>scalar values</a>. These
+<a>scalar values</a> can also be addressed as <a>code points</a>.
+
+<p class=note>A <a>scalar value string</a> is useful for any kind of I/O or other kind of operation
+where <a>UTF-8 encode</a> comes into play.
+<!-- It's also useful if you can imagine the subsystem to be implemented in Rust -->
+
+<p><dfn export lt=string>String</dfn> can be used to refer to either a <a>JavaScript string</a> or
+<a>scalar value string</a>, when it is clear from the context which is meant or when the distinction
+is immaterial. <a>Strings</a> are denoted by double quotes and monospace font.
 
 <p class=example id=example-string-notation>"<code>Hello, world!</code>" is a string.
 
+<p>To <dfn export for=string>cast</dfn> a <a>JavaScript string</a> into a
```

Comment:
Is there a less generic word for this procedure? Maybe one people are already using? I guess not since Web IDL has a very long phrase for it.

Maybe "scalarize"?